graphqlapi/src/main/java/io/stargate/graphql/schema/FieldFilterInputTypeCache.java

```diff
@@ -0,0 +1,140 @@
+package io.stargate.graphql.schema;
+
+import static graphql.schema.GraphQLList.list;
+
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputType;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLNonNull;
+import graphql.schema.GraphQLScalarType;
+import io.stargate.db.schema.Column;
+
+/**
+ * Caches GraphQL field "filter input" types, for example 'StringFilterInput' in:
+ *
+ * <pre>
+ * input BooksFilterInput {
+ *   author: StringFilterInput
+ *   title: StringFilterInput
+ * }
+ * </pre>
+ *
+ * These are used when receiving filtering criteria, for example in a 'books' query:
+ *
+ * <pre>
+ * query {
+ *   books(filter: { author: { eq: "Cormac McCarthy" } }) { ... }
+ * }
+ * </pre>
+ *
+ * Note that these filters are also used in "DELETE IF" and "UPDATE IF" conditions. Some operators
+ * only apply in one context, for example "notEq" is allowed in conditional updates, but not
+ * selects. We currently have a single filter type with all the operators, it's up to the user to a
+ * valid one otherwise the query will fail. We could possibly refine this by creating multiple
+ * filter types depending on the context (whether the column is a collection, is a PK, has an SAI
+ * index defined for it, etc).
+ */
+public class FieldFilterInputTypeCache extends FieldTypeCache<GraphQLInputType> {
+
+  private final FieldInputTypeCache inputTypeCache;
+
+  FieldFilterInputTypeCache(FieldInputTypeCache inputTypeCache, NameMapping nameMapping) {
+    super(nameMapping);
+    this.inputTypeCache = inputTypeCache;
+  }
+
+  @Override
+  protected GraphQLInputType compute(Column.ColumnType columnType) {
+    GraphQLInputType gqlInputType = inputTypeCache.get(columnType);
+    if (columnType.isMap()) {
+      return mapFilter(gqlInputType);
+    } else if (columnType.isList() || columnType.isSet()) {
+      return listFilter(gqlInputType);
+    } else {
```

Comment:
We don't need special UDT handling over here?

Reply:
Right now UDT filters look just like scalar filters:
```
input AUdtFilterInput {
  eq: AUdtInput
  gt: AUdtInput
  gte: AUdtInput
  in: [AUdtInput]
  lt: AUdtInput
  lte: AUdtInput
  notEq: AUdtInput
}
input IntFilterInput {
  eq: Int
  gt: Int
  gte: Int
  in: [Int]
  lt: Int
  lte: Int
  notEq: Int
}
```
So there is no special logic, we just pull an existing "simple" type from `inputTypeCache` and build upon it. 

That being said, I think there might be a UDT field predicate `WHERE a.i = 1` at least in Cassandra 4. I'm going to check and if that's the case I'll extend the basic filter like for collections.

Reply:
Hrmm yeah, field predicates are a thing in Cassandra 4 and DSE, they're not allowed in WHERE clauses but they are valid in IF conditions:
```
cqlsh:test> CREATE TYPE t(i int, s text);
cqlsh:test> CREATE TABLE foo(k int PRIMARY KEY, t t);
cqlsh:test> DELETE FROM foo WHERE k=1 IF t.i=2;
```
We can extend the UDT filter with a nested filter for each field:
```
input TUdtFilterInput {
  ... basic operators
  i: IntFilterInput
  s: StringFilterInput
}
```
Which would allow us to write something like:
```
deleteFoo(
  value: {k: 1},
  ifCondition: { t: { i: { eq: 2 } }
)
```
One catch is that a UDT could have fields with names that clash with basic operators: "eq", "gt", etc. We can add another level of nesting `{ t: { field: { i: { eq: 2 } } }`, but that complicates the syntax and we'll have to generate another graphql type.